### PR TITLE
update gobgp to 2.19.0

### PR DESCRIPTION
**Pull request by murali-reddy**

### CODE REVIEW

**Comment by aauren** [comment]
Thanks for starting this off @murali-reddy! Let me know when it's in a place for review.

**Comment by murali-reddy** [comment]
This PR is ready for review. I have done significant testing to cover below functional areas

- basic pod CIDR's of nodes are exchanges between the iBGP nodes
- routes to pod CIDR's are installed to local FIB based on the updates to the gobgp RIB
- service VIP's are advertised to external peers
- appropriate BGP import and export policies as required by kube-router are configured
- all prefix sets and neighbor sets used by BGP policies are populated and updated as the nodes, services gets addedd/deleted
- route reflector configuration

Also all unit tests are modified as per the gobgp 2.* API's and are passing.

File: pkg/controllers/routing/bgp_peers.go [hunk 1]
```diff
@@ -144,22 +141,18 @@ func (nrc *NetworkRoutingController) syncInternalPeers() {
...
-						RouteReflectorClusterId: config.RrClusterIdType(nrc.bgpClusterID),
-					},
+				n.RouteReflector = &gobgpapi.RouteReflector{
+					RouteReflectorClient:    true,
+					RouteReflectorClusterId: fmt.Sprint(nrc.bgpClusterID),
 				}
 			}
 		}
 
 		// TODO: check if a node is alredy added as nieighbour in a better way than add and catch error
```
Comment (mrueg) [inline pkg/controllers/routing/bgp_peers.go:151]: Is that TODO still required? (also there's a small typo in neighbor)

File: pkg/controllers/routing/bgp_policies.go [hunk 2]
```diff
@@ -1,141 +1,345 @@
...
-	})
-	err = nrc.bgpServer.ReplaceDefinedSet(ns)
-	if err != nil {
-		err2 := nrc.bgpServer.AddDefinedSet(ns)
-		if err2 != nil {
-			glog.Errorf("Failed to add ns: %s", err2)
+	return externalBGPPeerCIDRs, nil
+}
+
+// a slice of all peers is used as a match condition for reject statement of servicevipsdefinedset import polcy
```
Comment (mrueg) [inline pkg/controllers/routing/bgp_policies.go:282]: typo: policy

File: pkg/controllers/routing/network_routes_controller.go [hunk 3]
```diff
@@ -714,18 +750,14 @@ func (nrc *NetworkRoutingController) startBgpServer() error {
 		nrc.pathPrependCount = uint8(repeatN)
 	}
 
-	nrc.bgpServer = gobgp.NewBgpServer()
+	//nrc.bgpServer = gobgp.NewBgpServer()
```
Comment (mrueg) [inline pkg/controllers/routing/network_routes_controller.go:753]: that can be removed?

**Comment by murali-reddy** [comment]
For a referance the exposed consumable API from gobgp are present in

https://github.com/osrg/gobgp/blob/master/api/gobgp.pb.go
https://github.com/osrg/gobgp/blob/master/api/gobgp.proto

**Comment by murali-reddy** [comment]
All the unit tests and lint errors fixed. https://travis-ci.org/github/cloudnativelabs/kube-router/builds/723788806  Addresses review comments as well.

**Pull request merged**